client/my-sites/themes/controller.js

```diff
@@ -125,10 +126,11 @@ export function details( context, next ) {
 		isLoggedIn: !! user
 	};
 
-	context.store.dispatch( setSection( 'themes', {
-		hasSidebar: false,
+	// When we're logged in, we need to remove the sidebar.
+	removeSidebar( context, {
+		section: 'themes',
 		isFullScreen: true
-	} ) );
+	} );
```

Comment:
> @ockham, is there a chance to include the `context.store.dispatch()` call in `LayoutLoggedOut`? This will make sure we don't have any low-level dispatch calls in non-layout/non-library code.

We could wrap it in a middleware, or actually, a generic factory to produce one, i.e. something akin to this idea: https://github.com/Automattic/wp-calypso/pull/3425/files?diff=unified#diff-85f90b61d9c36634c5464a3180470726R108